### PR TITLE
Fix compressed popup when reaching right bound of map

### PR DESCRIPTION
**Pull request by xintongxia**
For https://github.com/uber/react-map-gl/issues/676

### CODE REVIEW

**Comment by Pessimistress** [comment]
If I'm reading this correctly, it doesn't fix anything, just offers an option to fix it?

**Comment by xintongxia** [comment]
> If I'm reading this correctly, it doesn't fix anything, just offers an option to fix it?

I added a prop `minWidth`

**Comment by Pessimistress** [comment]
That is a breaking change and we can't publish it in a patch release.
- Does mapbox apply minWidth to their popups, if so, what's the value?
- Add to what's new and upgrade guide

File: src/components/popup.js [hunk 1]
```diff
@@ -131,14 +135,17 @@ export default class Popup extends BaseControl<PopupProps, *, HTMLDivElement> {
...
     const style = {
       position: 'absolute',
-      left: x + offsetLeft,
-      top: y + offsetTop,
-      transform: `translate(${-anchorPosition.x * 100}%, ${-anchorPosition.y * 100}%)`,
+      transform: `
+        translate(${-anchorPosition.x * 100}%, ${-anchorPosition.y * 100}%)
+        translate(${left}px, ${top}px)
+      `,
+      maxWidth,
```
Comment (xintongxia) [inline src/components/popup.js]: https://github.com/mapbox/mapbox-gl-js/blob/master/src/ui/popup.js#L406

File: src/components/popup.js [hunk 2]
```diff
@@ -40,6 +40,8 @@ const propTypes = Object.assign({}, BaseControl.propTypes, {
   offsetLeft: PropTypes.number,
   // Offset from the top
   offsetTop: PropTypes.number,
+  // A number that sets the CSS property of the popup's maxWidth in pixels
+  maxWidth: PropTypes.number,
```
Comment (xintongxia) [inline src/components/popup.js]: https://github.com/mapbox/mapbox-gl-js/blob/master/src/ui/popup.js#L31

**Pull request merged**